File: frontend/src/types/HazardTicket.ts
export interface HazardTicket {
  id: number;
  result_id: number;
  device_id: number;
  item_code: string;
  severity: string;
  owner_id: number;
  deadline: string;
  rectify_status: string;
  rectify_note: string;
  rectify_photo_url: string;
  merged_result_ids: number[];
  rectified_at: string;
  recheck_note: string;
  closed_at: string;
}

File: backend/src/models/hazard_ticket.py
from pydantic import BaseModel


class HazardTicket(BaseModel):
    id: int | float
    result_id: int | float
    device_id: int | float
    item_code: str
    severity: str
    owner_id: int | float
    deadline: str
    rectify_status: str
    rectify_note: str
    rectify_photo_url: str
    merged_result_ids: list = []
    rectified_at: str = ""
    recheck_note: str = ""
    closed_at: str = ""
